mapserver/django/django_project/fba/management/commands/ingest_boundary_data.py

```diff
@@ -119,35 +201,91 @@ def handle(self, *args, **options):
+                                admin_code_key[target_table_name]: admin_code,
+                                'defaults': {
+                                    'id': self.get_last_id(ModelClass),
+                                    'geom': geom,
+                                    'name': name,
+                                }
+                            }
+                            boundary, created = ModelClass.objects.get_or_create(
+                                **data)
+                            if not created:
+                                del data['defaults']['id']
+                                ModelClass.objects.update_or_create(**data)
+
+                            print(f'{target_table_name.capitalize()} '
+                                  f'{schema_name} - {name} = {created}')
                         except Exception as e:
                             print(str(e))
+                            raise e
+
+                    # If it has population column, must update census table
+                    if has_pop_column:
+                        if adm_level != 3:
+                            # population data is not on village level.
+                            # Interpolate/Assume/Use current level to the
+                            # lower level
+                            for level in range(adm_level + 1, 4):
+                                target_table_name = target_table_mapping[level]
+                                ModelClass = Models[target_table_name]
+                                admin_code = self.generate_admin_code(
+                                    schema_name, level, origin_id)
+                                data = {
+                                    admin_code_key[
+                                        target_table_name]: admin_code,
+                                    'defaults': {
+                                        'id': self.get_last_id(ModelClass),
+                                        'geom': geom,
+                                        'name': name
+                                    }
+                                }
+                                boundary, created = ModelClass.objects.get_or_create(
+                                    **data)
+                                if not created:
+                                    del data['defaults']['id']
+                                    ModelClass.objects.update_or_create(**data)
+
+                        # boundary must now refer to village level
+                        # insert population data to census table
+                        data = {
+                            'village_id': boundary.village_code,
+                            'defaults': {
+                                'population': pop
+                            }
+                        }
+                        census_entry, created = Census.objects.update_or_create(
+                            **data)
```

Comment:
We should refactor it. The code logic has become too long. Ideally extract this into utils and call it in the management scripts.